lib/evmone/instructions.hpp

```diff
@@ -820,6 +820,40 @@ inline void swap(StackTop stack) noexcept
     a[3] = t3;
 }
 
+inline code_iterator dupn(StackTop stack, ExecutionState& state, code_iterator pos) noexcept
+{
+    const auto n = pos[1];
+
+    const auto stack_size = &stack.top() - state.stack_space.bottom();
+
+    if (stack_size <= n)
+    {
+        state.status = EVMC_STACK_UNDERFLOW;
+        return nullptr;
+    }
+
+    stack.push(stack[n]);
```

Comment:
I think this misses the `< 1024` check. `push` does [not seem enforce the 1024 limit](https://github.com/ethereum/evmone/blob/master/lib/evmone/instructions.hpp#L38).

Reply:
The stack overflow check is done automatically by the interpreter. This information is in the instruction traits.